lib/core/src/lib/common/utils/date-fns-utils.ts

```diff
@@ -78,4 +79,72 @@ export class DateFnsUtils {
         }
         return dateFnsLocale;
     }
+
+    /**
+     * A mapping of Moment.js format tokens to date-fns format tokens.
+     */
+    static momentToDateFnsMap = {
+        M: 'M',
+        D: 'd',
+        Y: 'y',
+        A: 'a'
+    };
+
+    /**
+     * A mapping of date-fns format tokens to Moment.js format tokens.
+     */
+    static dateFnsToMomentMap = {
+        M: 'M',
+        d: 'D',
+        y: 'Y',
+        a: 'A'
+    };
+
+    /**
+     * Converts a Moment.js date format string to the equivalent date-fns format string.
+     *
+     * @param dateDisplayFormat - The Moment.js date format string to convert.
+     * @returns The equivalent date-fns format string.
+     */
+    static convertMomentToDateFnsFormat(dateDisplayFormat: string): string {
```

Comment:
add safety checks for all the functions, if the input parameter is null or empty

Reply:
added checks